atomic_reactor/plugins/post_export_operator_manifests.py

```diff
@@ -0,0 +1,179 @@
+            raise RuntimeError('Provided image is not a tar archive')
+
+        try:
+            image_archive = tarfile.TarFile(image_archive_path)
+        except tarfile.ReadError as ex:
+            self.log.error("Could not read image archive: %s" % ex)
+            raise RuntimeError("Could not read image archive: %s" % ex)
+
+        try:
+            manifest_info = image_archive.getmember(IMAGE_MANIFEST_NAME)
+        except KeyError as ex:
+            raise RuntimeError('Image archive format error. Manifest file not found %s' % ex)
+
+        if not manifest_info.isfile():
+            raise
+
+        manifest_buffer = image_archive.extractfile(manifest_info)
+        manifest = json.load(manifest_buffer)
+
+        for image in manifest:
+            # We rely on the order of the array in manifests.json
+            for layer_path in image['Layers']:
+                try:
+                    layer_archive_info = image_archive.getmember(layer_path)
+                except KeyError as ex:
+                    raise RuntimeError('Image archive format error. '
+                                       'Layer archive not found for %s: %s' % (layer_path, ex))
+                if not layer_archive_info.isfile():
+                    raise RuntimeError('No layer available for %s' % layer_path)
+                # layers can get large, let's extract them in the filesystem instead of load them
+                if layer_path.startswith('/') or '..' in layer_path:
+                    raise RuntimeError('Unsafe path found in image archive')
+                self.log.debug("extracting %s", layer_archive_info.name)
+                image_archive.extract(layer_archive_info, workdir)
+
+                layer_archive_path = os.path.join(workdir, layer_archive_info.name)
+                layer_archive = tarfile.TarFile(layer_archive_path)
+                layer_archive.extractall(workdir, self.get_archived_manifests(layer_archive))
+
+                layer_dir_path = layer_archive_info.name.split('/')[0]
+                layer_dir_path = os.path.join(workdir, layer_dir_path)
+                self.log.debug("removing %s", layer_dir_path)
+                shutil.rmtree(layer_dir_path)
+
+                remove_list = []
+                for entry in os.listdir(workdir):
+                    if entry.startswith(WHITEOUT_FILE_PREFIX):
+                        self.log.debug("%s found. Marking %s for removal",
+                                       entry, entry[len(WHITEOUT_FILE_PREFIX):])
+                        remove_list.append(entry)
+                        remove_list.append(entry[len(WHITEOUT_FILE_PREFIX):])
+                for entry in remove_list:
+                    os.unlink(entry)
+
+        manifests_path = os.path.join(workdir, MANIFESTS_DIR_NAME)
+        if not os.path.isdir(manifests_path):
+            self.log.warning('No operator manifest files found. No archive will be created.')
```

Comment:
IMO this should be error, if operator image is declared by label, it should contain `/manifests`.

Reply:
Agreed.  This should fail the build if the label is set but `manifests_path` is non-existent *or empty*.